frontend-react/src/components/USLink.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { PropsWithChildren } from "react";
+import { Link, NavLink } from "react-router-dom";
+
+// Known issue with the `PropsWithChildren generic in React 18,
```

Comment:
This is still an issue if we use function declarations for components?  We also might not even need to extend `PropsWithChildren` in the first place if we just declare `children` as a prop on its own 🤷 

```jsx
interface USNavLinkProps {
    href: string;
    className?: string;
    activeClassName?: string;
    children?: React.ReactNode
}

export function USLink({ children, href, className }: USNavLinkProps) {
  return (...);
}
```

Reply:
PropsWithChildren is the preferred way to extend props with React so they manage what "child" means and we remain current; the ideal method is packing our interface custom shape into the generic, but I did it like this in case we moved to React 18 before they patched that bug. 😩

Reply:
But so far as function declaration vs jsx vs text... ReactNode covers it all so PropsWithChildren should be fine here in that regard

Reply:
Right, but I'm wondering why we even need to use `PropsWithChildren` in the first place since it's just a union between the provided generic type and an object with an optional `ReactNode` for `children`: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L800.  It feels like listing `children` directly in our prop type/interface would be clearer/less circuitous.

Reply:
Yeah, it would reduce the need for import and such, but my mind goes to the points of change; if React changes everywhere we used it changes too, but we lose that if we define it ourselves. 

Let's generalize this and consider it in norming! Something like "when do we rely on our packages types vs. defining our own, and why?" 🙂 I think "why" is the sauce here; I'm sure there's stuff we aren't thinking of.